debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -244,18 +248,21 @@ protected void readChunk() throws InterruptedException {
             context.startNewChunk();
             emitWindowOpen();
             while (context.snapshotRunning()) {
-                final TableId currentTableId = (TableId) context.currentDataCollectionId();
-                currentTable = databaseSchema.tableFor(currentTableId);
-                if (currentTable == null) {
-                    LOGGER.warn("Schema not found for table '{}', known tables {}", currentTableId, databaseSchema.tableIds());
-                    nextDataCollection();
+                if (isTableInvalid()) {
                     continue;
                 }
-                if (currentTable.primaryKeyColumns().isEmpty()) {
-                    LOGGER.warn("Incremental snapshot for table '{}' skipped cause the table has no primary keys", currentTableId);
-                    nextDataCollection();
-                    continue;
+                if (connectorConfig.supportsSchemaChangesDuringIncrementalSnapshot()) {
+                    if (!context.isSchemaVerified()) {
+                        verifySchema();
+                        if (isTableInvalid()) {
+                            continue;
+                        }
+                        if (!context.isSchemaVerified()) {
+                            break;
```

Comment:
When this situation happens and what should be the action?

Reply:
Agreed it's confusing. Added a comment